libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/maneuver/ManeuverState.kt

```diff
@@ -0,0 +1,30 @@
+package com.mapbox.navigation.dropin.component.maneuver
+
+import com.mapbox.bindgen.Expected
+import com.mapbox.bindgen.ExpectedFactory
+import com.mapbox.navigation.base.ExperimentalMapboxNavigationAPI
+import com.mapbox.navigation.dropin.component.navigationstate.NavigationState
+import com.mapbox.navigation.ui.maneuver.model.Maneuver
+import com.mapbox.navigation.ui.maneuver.model.ManeuverError
+import com.mapbox.navigation.ui.maneuver.model.ManeuverErrorFactory
+
+data class ManeuverState(
+    val isVisible: Boolean,
+    val navigationState: NavigationState,
+    val maneuver: Expected<ManeuverError, List<Maneuver>>
```

Comment:
Here only `isVisible` and `maneuver` are actual result values which are then used to update the UI. `navigationState` is a data point out of which the result is computed.

We might want to separate these to have 2 object, one object that holds `isVisible` and `maneuver` state, and the other that holds previous input values for reprocessing when values are updated.

Reply:
Same in all other components.

Reply:
I am wondering what is the advantage of separating then out into separate components?

Reply:
Readability in my opinion. It wasn't immediately clear to me why do we store the `navigationState` here since it is not used for any view operations directly.

I understand that it's here to be able to reduce the state when new route progress comes in but it might be worth decoupling the actual state that views feed off of from the internal view model's state.

Reply:
Not a very strong point here though, just my small observation. We can address this later as we add more state properties and see what fits where.

Reply:
Yep. I believe we are still not clear on what other data points would be required for our components. It would probably make more sense to update this if required once we have a good amount of functionality ready so then we have more clear understanding. For ex: we are at the moment not tracking any UI state like storing the recycler view scroll position that would help to show the recycler view from the position you left it.